build.sbt

```diff
@@ -45,6 +45,7 @@ val commonSettings = Seq(
   libraryDependencies ++= Seq(
         Libraries.catsEffectKernel,
         Libraries.redisClient,
+        Libraries.keyPool         % Optional,
```

Comment:
hey @BusyByte, is there any particular reason this is added as Optional? It's a bit of a footgun, looks like we've just spotted it causing trouble in: https://github.com/profunktor/redis4cats/discussions/955

How about we just move the dependency to the effects module, like this: https://github.com/profunktor/redis4cats/pull/956 ?